src/EditorFeatures/Core/Implementation/ExtractInterface/AbstractExtractInterfaceCommandHandler.cs

```diff
@@ -70,11 +79,12 @@ public bool ExecuteCommand(ExtractInterfaceCommandArgs args, CommandExecutionCon
             // and also will take it into consideration when measuring command handling duration.
             context.OperationContext.TakeOwnership();
             var extractInterfaceService = document.GetLanguageService<AbstractExtractInterfaceService>();
-            var result = extractInterfaceService.ExtractInterface(
-                document,
-                caretPoint.Value.Position,
-                (errorMessage, severity) => workspace.Services.GetService<INotificationService>().SendNotification(errorMessage, severity: severity),
-                CancellationToken.None);
+            var result = _threadingContext.JoinableTaskFactory.Run(async () =>
+                await extractInterfaceService.ExtractInterfaceAsync(
+                    document,
+                    caretPoint.Value.Position,
+                    (errorMessage, severity) => workspace.Services.GetService<INotificationService>().SendNotification(errorMessage, severity: severity),
+                    CancellationToken.None).ConfigureAwait(false));
```

Comment:
why do you need the 'async/await/configurewait' here?  For that matter, can you document what's going on with the use of JTF?  thanks!

Reply:
I'm going to defer to @heejaechang  to help explain this. 

From my understanding, take with grain of salt: 

JTF is needed because command handler is going to be called from UI thread. 
async/await to get the correct signature for lambda in JTF. 
configureawait because we don't actually need it to synchronize with any thread. 

Hopefully helpful. This seemed to be one of the things I was missing in my first attempt to make everything async.